### PR TITLE
Report UDP network statistics

### LINKED ISSUE #1093: Add metrics for udp packet received/errors
See 
https://github.com/solana-labs/solana/issues/1091#issuecomment-417161573

### PR DESCRIPTION
**Pull request by pgarg66**
Fixes #1093

### CODE REVIEW

File: scripts/net-stats.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+source configure-metrics.sh
+
+[[ $(uname) == Linux ]] || exit 0
+while true; do
+  net_stat=$(netstat -suna)
+  packets_received=$(echo "$net_stat" | awk 'BEGIN {tmp_var = 0} /packets received/ {tmp_var = $1} END { print tmp_var }')
+  receive_errors=$(echo "$net_stat" | awk 'BEGIN {tmp_var = 0} /packet receive errors/ {tmp_var = $1} END { print tmp_var }')
+  rcvbuf_errors=$(echo "$net_stat" | awk 'BEGIN {tmp_var = 0} /RcvbufErrors/ {tmp_var = $2} END { print tmp_var }')
+
+  echo "packets_received=$packets_received,receive_errors=$receive_errors,rcvbuf_errors=$rcvbuf_errors"
```
Comment (mvines) [inline scripts/net-stats.sh]: nit: assign to a variable then reuse in the next line to avoid duplication

File: scripts/net-stats.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#
+# Reports network statistics
+#
+
+cd "$(dirname "$0")"
+
+# shellcheck source=scripts/configure-metrics.sh
+source configure-metrics.sh
+
+[[ $(uname) == Linux ]] || exit 0
```
Comment (mvines) [inline scripts/net-stats.sh]: @pgarg66 and I talked in IRL about sending the diff since the last netstat run rather than the abs value

**Comment by aeyakovenko** [comment]
@mvines can we add these to graphana?

**Comment by mvines** [comment]
Yep.  As a part of my devops strike week, I'm very happy to pair with anybody who wants to add these to grafana!

File: scripts/net-stats.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+packets_received=0
+packets_received_diff=0
+receive_errors=0
+receive_errors_diff=0
+rcvbuf_errors=0
+rcvbuf_errors_diff=0
+
+update_netstat() {
+  net_stat=$(netstat -suna)
```
Comment (mvines) [inline scripts/net-stats.sh]: nit: `declare net_stat=$(netstat -suna)`

File: scripts/net-stats.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
```
Comment (mvines) [inline scripts/net-stats.sh]: nit: `declare stats` too, to keep it local

File: scripts/net-stats.sh [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  receive_errors="$stats"
+
+  stats=$(echo "$net_stat" | awk 'BEGIN {tmp_var = 0} /RcvbufErrors/ {tmp_var = $2} END { print tmp_var }')
+  rcvbuf_errors_diff=$(( stats-rcvbuf_errors ))
+  rcvbuf_errors="$stats"
+}
+
+update_netstat
+
+[[ $(uname) == Linux ]] || exit 0
```
Comment (mvines) [inline scripts/net-stats.sh]: let's move the Linux check to line 5 IMO, but definitely before line 34

**Comment by garious** [comment]
@pgarg66, fyi, you made the yaml file executable. Would be cool to figure out why those bits keep flipping for you. Maybe there's something you can do with your git configuration to prevent that.

**Comment by pgarg66** [comment]
My samba mounted filesystem (Linux to Mac) that’s causing these problem for me. I generally correct the permissions for the rust files before pushing them to PR.

I forgot about yaml. I’ll fix it in the next PR.

**Pull request merged**